File: src/main/java/com/example/demo/util/TimesheetGenerator.java
package com.example.demo.util;

import lombok.experimental.UtilityClass;

import java.time.LocalDate;
import java.time.YearMonth;
import java.util.ArrayList;
import java.util.List;
import java.util.Locale;

@UtilityClass
public class TimesheetGenerator {

    public List<LocalDate> generateDates(YearMonth yearMonth) {
        int days = yearMonth.lengthOfMonth();
        List<LocalDate> dates = new ArrayList<>(days);
        for (int day = 1; day <= days; day++) {
            dates.add(yearMonth.atDay(day));
        }
        return dates;
    }

    public String formatYearMonth(YearMonth yearMonth) {
        return formatYearMonth(yearMonth, Locale.JAPAN);
    }

    public String formatYearMonth(YearMonth yearMonth, Locale locale) {
        return String.format(locale, "%d年%d月", yearMonth.getYear(), yearMonth.getMonthValue());
    }
}

File: src/main/kotlin/com/example/demo/util/TimesheetGenerator.kt
package com.example.demo.util

import java.time.LocalDate
import java.time.YearMonth
import java.util.*

object TimesheetGenerator {
    fun generateDates(yearMonth: YearMonth): List<LocalDate> =
        (1..yearMonth.lengthOfMonth()).map { day -> yearMonth.atDay(day) }

    fun formatYearMonth(yearMonth: YearMonth, locale: Locale = Locale.JAPAN): String =
        "%d年%d月".format(locale, yearMonth.year, yearMonth.monthValue)
}

